extension/jaegerremotesampling/internal/remote_strategy_cache.go

```diff
@@ -0,0 +1,123 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+
+import (
+	"sync"
+	"time"
+
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+)
+
+type serviceStrategyCache interface {
+	get(serviceName string) (*sampling.SamplingStrategyResponse, bool)
+	put(serviceName string, response *sampling.SamplingStrategyResponse)
+	close() error
```

Comment:
whoops, addressing this now. after rebasing I managed to actually throw away the invocation of this closer, so I'm glad this came up.

BTW: since we use the Jaeger-defined `strategystore.StrategyStore` interface for both our remote/local variations, I'm going to just be returning this cache now that it's an `io.Closer` for easy access (to close during component lifecycle) in `extension.go`